main.cpp

```diff
@@ -76,9 +76,660 @@ Purpose:  This project continues developing Project3.
 
  Wait for my code review.
  */
-
 #include <iostream>
+#include <string>
+
+struct PencilCase
+{
+
+	char openMechanism = 'Z';
+	int maxItemsToHold;
+	bool isOpen;
+	int numItemsAdded = 0;
+    bool inSchoolBag = false;
+
+	PencilCase();
+	~PencilCase();
+
+	void open(); 
+	void close();	
+    void addItems(int itemsToAdd = 1);
+    void removeItems(int itemsToRemove);
+	
+    struct Pencil
+	{
+		float sharpness = 1.5f;
+		bool sharpEnough = false;
+		std::string color = "Brown";
+		bool canColor;
+		std::string shade;
+		
+		Pencil();
+		~Pencil();
+		
+		void sharpen(int rotations);
+		void writeSomething(std::string writing);
+		bool isSharp();
+	};
+};
+
+struct Drill
+{
+	bool isPlugged;
+	bool drillbitAttached;
+	int powerInWatts;
+	char drillType = 'H';
+	float rotationsToSecureBit = 0;
+	float chuckTightness = 0.f;
+
+	Drill();
+	~Drill();
+
+	void drillHole(int holes);
+	void unplug();
+	void attachDrillbit(float drillbitWidth);
+};
+
+struct Elevator
+{
+
+	int highestFloor = 50;
+	int lowestFloor = -2;
+	double totalWeightCarried = 0.0;
+	bool elevatorArrived = false;
+	double maxWeight;
+	int currentFloor;
+	double lastUserWeight = 0;
+	std::string name;
+
+	Elevator(std::string name);
+	~Elevator();
+
+	void goToFloor(int designatedFloor);
+    void onboardPassenger(double userWeight);
+	void offboardPassenger();
+	bool canTakePassengerWeight(double userWeight);
+	void updateWeightCarried(double userWeight, bool userExit);
+	bool arrived();
+	
+    struct ElevatorUser
+	{
+		double weight;
+		int destinationFloor;
+		int currentFloor;
+		bool isInsideElevator = false;
+		bool canExit = true;
+
+		ElevatorUser();
+		~ElevatorUser();
+
+		void callElevator();
+        void enterElevator();
+        void setDestination(int newDestinationFloor);
+		void exitElevator();
+		void takeStairsInstead(int newDestinationFloor);
+	};
+};
+
+struct SchoolBag
+{
+    PencilCase schoolPencilCase;
+
+    SchoolBag();
+    ~SchoolBag();
+
+    void packPencilCase();
+    void unpackPencilCase();
+};
+
+struct Building
+{
+    Elevator elevatorA{"A1"};
+    Elevator elevatorB{"B1"};
+	bool isOpen;
+
+    Building();
+    ~Building();
+
+	void closeBuilding();
+	void openBuilding();
+	bool buildingEmpty();
+	void clearBuilding();
+	void callElevatorAAndTakePassenger(int designatedFloor, double passengerWeight);
+};
```

Comment:
Fix this indentation
- [ ] revise

Reply:
Copying the code from Visual Studio caused this. On replit the code is correctly indented.